src/components/icon/icon.ts

```diff
@@ -71,14 +71,22 @@ export default class IgcIconComponent extends SizableMixin(LitElement) {
   @property({ type: Boolean, reflect: true })
   public mirrored = false;
 
+  /**
+   * The role of icon element.
+   * Defaults to `img`.
+   *
+   * @attr [role="img"]
+   */
+  @property({ reflect: true })
+  public role = 'img';
+
   constructor() {
     super();
     this.size = 'medium';
   }
 
   public override connectedCallback() {
     super.connectedCallback();
-    this.setAttribute('role', 'img');
```

Comment:
Can we check for an existing role attribute and then set the default if none is found?

Something about a public property used in an internal context that is reflected as a standard attribute feels off to me.